pkg/util/validation/validate.go

```diff
@@ -61,6 +61,29 @@ const (
 	TooManyHAClusters = "too_many_ha_clusters"
 )
 
+// allDiscardedSampleReasons is used when cleaning up metrics for a user.
+var allDiscardedSampleReasons = []string{
```

Comment:
We also have reasons defined in the ingester (sampleOutOfBounds, sampleOutOfOrder, newValueForTimestamp). The ingester also does this `validation.DiscardedSamples.WithLabelValues(ve.errorType, userID).Inc()` which we need to check all possible values there.

Reply:
Good point, thanks!